src/main/java/gregtech/common/tileentities/machines/basic/GT_MetaTileEntity_Miner.java

```diff
@@ -0,0 +1,202 @@
+package gregtech.common.tileentities.machines.basic;
+
+import gregtech.api.enums.Textures;
+import gregtech.api.interfaces.ITexture;
+import gregtech.api.interfaces.metatileentity.IMetaTileEntity;
+import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
+import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_BasicMachine;
+import gregtech.api.objects.GT_RenderedTexture;
+import gregtech.api.objects.ItemData;
+import gregtech.api.util.GT_ModHandler;
+import gregtech.api.util.GT_OreDictUnificator;
+import gregtech.api.util.GT_Utility;
+import gregtech.common.blocks.GT_Block_Ores_Abstract;
+import gregtech.common.blocks.GT_TileEntity_Ores;
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+
+import java.util.ArrayList;
+
+public class GT_MetaTileEntity_Miner extends GT_MetaTileEntity_BasicMachine {
+    private static final ItemStack miningPipe = GT_ModHandler.getIC2Item("miningPipe", 0);
+    private static final Block miningPipeBlock = GT_Utility.getBlockFromStack(miningPipe);
+    private static final Block miningPipeTipBlock = GT_Utility.getBlockFromStack(GT_ModHandler.getIC2Item("miningPipeTip", 0));
+
+    int drillX, drillY, drillZ;
+    boolean isPickingPipes;
+    boolean waitMiningPipe;
+    static int[] radius = new int[]{8, 8, 24, 40}; //Miner radius per tier
+    static int[] speed = new int[]{20, 20, 5, 2}; //Miner cycle time per tier
+    static int[] energy = new int[]{24, 24, 96, 384}; //Miner energy consumption per tier
```

Comment:
Well yes `static final` should be named an upper case. But I thought upper case names mean that they are constant and **value type** variables. MINING_PIPE_BLOCK is a reference type variable and because of this I asked.